app/code/core/Mage/Sales/Model/Order.php

```diff
@@ -1895,8 +1895,10 @@ public function getStatusHistoryCollection($reload = false)
                 ->setOrder('entity_id', 'desc');
 
             if ($this->getId()) {
+                /** @var Mage_Sales_Model_Order_Status_History $status */
                 foreach ($this->_statusHistory as $status) {
                     $status->setOrder($this);
+                    $status->setDataChanges(false);
```

Comment:
it seems we need to add this setDataChanges calls in a lot of places, but doesn't it mean we should modify some Abstract or something like that to solve this issue in a more general way?